src/java/org/apache/cassandra/net/StartupClusterConnectivityChecker.java

```diff
@@ -48,81 +51,133 @@
 {
     private static final Logger logger = LoggerFactory.getLogger(StartupClusterConnectivityChecker.class);
 
-    private final int targetPercent;
+    private final boolean blockForRemoteDcs;
     private final long timeoutNanos;
 
-    public static StartupClusterConnectivityChecker create(int targetPercent, int timeoutSecs)
+    public static StartupClusterConnectivityChecker create(long timeoutSecs, boolean blockForRemoteDcs)
     {
-        timeoutSecs = Math.max(1, timeoutSecs);
+        if (timeoutSecs < 0)
+            logger.warn("skipping block-for-peers due to negative timeout. You may encounter errors or timeouts on" +
```

Comment:
Should this be a warning? They have pretty clearly asked for it?

Reply:
I thought it should be for the same reason we warn them if they set the timeout to more than 100, but I don't care too much. I removed it.

Reply:
The reason I thought it was strange was because -1 was a valid config option.